### PR TITLE
Add test of oneDPL interfaces

### PR DESCRIPTION
**Pull request by timmiesmith**
Test calls interfaces whose names changed in Beta process.

Signed-off-by: Timmie Smith <timmie.smith@intel.com>

### CODE REVIEW

File: test/extensions_testsuite/interface_check.pass.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
+// -*- C++ -*-
+//===-- interface_check.pass.cpp --------------------------------------------===//
+//
+// Copyright (C) 2020 Intel Corporation
```
Comment (capatober) [inline test/extensions_testsuite/interface_check.pass.cpp:4]: 2020 -> 2021

File: test/extensions_testsuite/interface_check.pass.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+#endif
+
+
+using oneapi::dpl::counting_iterator;
+using oneapi::dpl::discard_iterator;
+using oneapi::dpl::make_zip_iterator;
+using oneapi::dpl::begin;
+using oneapi::dpl::end;
+using oneapi::dpl::identity;
+using std::get;
```
Comment (capatober) [inline test/extensions_testsuite/interface_check.pass.cpp:47]: `using std::get` is not needed now

**Review by capatober — APPROVED** [review]

File: test/extensions_testsuite/interface_check.pass.cpp [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+using oneapi::dpl::identity;
+
+int main()
+{
+    cl::sycl::buffer<int, 1> buf{ cl::sycl::range<1>(10) };
+
+    auto b = begin(buf);
+    auto e = end(buf);
+    auto z = make_zip_iterator(counting_iterator<int>(), discard_iterator());
+    std::get<1>(z[0]) = identity()(*counting_iterator<int>());
```
Comment (capatober) [inline test/extensions_testsuite/interface_check.pass.cpp:55]: I think it's better mention `oneapi::dpl` explicitly here to not have a potential ambiguity with `std::begin`, `std::end`, `std::identity`. Then corresponding `using`s are unneeded.

**Review by capatober — APPROVED** [review]

**Comment by timmiesmith** [comment]
Closing this MR. New test will be added as part of refactoring in #81 so it is in the correct location.

**Pull request closed without merging**